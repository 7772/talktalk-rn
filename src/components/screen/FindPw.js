// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  ScrollView,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

type Styles = {
  container: ViewStyle,
  scrollView: ViewStyle,
  scrollViewContainer: ViewStyle,
  wrapper: ViewStyle,
  btnWrapper: ViewStyle,
  btnRegister: ViewStyle,
  txtRegister: TextStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    flexDirection: 'column',
    alignItems: 'center',
  },
  scrollView: {
    alignSelf: 'stretch',
  },
  scrollViewContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  wrapper: {
    marginTop: 40 * ratio,
    width: '78%',

    flexDirection: 'column',
    alignItems: 'center',
  },
  btnWrapper: {
    width: '100%',
    alignItems: 'flex-end',
  },
  btnRegister: {
    backgroundColor: colors.dodgerBlue,
    borderColor: colors.dodgerBlue,
    borderRadius: 4 * ratio,
    borderWidth: 1 * ratio,
    width: 136 * ratio,
    height: 60 * ratio,
    marginLeft: 4 * ratio,
    marginTop: 24 * ratio,
    marginBottom: 48 * ratio,
    shadowColor: colors.dodgerBlue,
    shadowOffset: {
      width: 0,
      height: 10 * ratio,
    },
    shadowRadius: 4 * ratio,
    shadowOpacity: 0.3,

    alignItems: 'center',
    justifyContent: 'center',
  },
  txtRegister: {
    fontSize: 16 * ratio,
    fontWeight: 'bold',
    color: 'white',
  },
});

type Props = {
  navigation: any,
};
type State = {
  email: string,
  isLoading: boolean,
};

class Screen extends Component<Props, State> {
  static navigationOptions = {
    title: getString('FIND_PW'),
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      isLoading: false,
      email: '',
    };
  }

  render() {
    return (
      <View style={styles.container}>
        <StatusBar isDarkContent={false}/>
        <ScrollView
          style={styles.scrollView}
          contentContainerStyle={styles.scrollViewContainer}
        >
          <View style={styles.wrapper}>
            <TextInput
              style={{ marginTop: 8 * ratio }}
              txtLabel={getString('EMAIL')}
              txtHint={ getString('EMAIL') }
              txt={ this.state.email }
              onTextChanged={ (text) => this.onTextChanged('EMAIL', text)}
              isPassword={ true }
            />
            <View style={styles.btnWrapper}>
              <Button
                id='send_link'
                isLoading={this.state.isLoading}
                onPress={() => this.onSendLink()}
                style={styles.btnRegister}
                textStyle={styles.txtRegister}
              >{getString('SEND_LINK')}</Button>
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }

  onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        this.setState({ email: text });
        break;
    }
  }

  goBack = () => {
    this.props.navigation.goBack();
  }

  onSendLink = () => {
    console.log('onSendLink');
    this.setState({
      isLoading: true,
    }, async() => {
      this.setState({ isLoading: false });
    });
  }
}

export default Screen;
