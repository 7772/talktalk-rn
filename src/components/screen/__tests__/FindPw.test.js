import 'react-native';
import * as React from 'react';
import FindPw from '../FindPw';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('rendering test', () => {
  const wrapper = shallow(
    <FindPw />,
  );

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('interaction', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
      },
    };
    wrapper = shallow(<FindPw {...props} />);
  });
  describe('clicking the button', () => {
    it('should call onSendLink callback', () => {
      const spy = jest.spyOn(wrapper.instance(), 'onSendLink');
      const btn = wrapper.find('#send_link');
      btn.props().onPress();
      expect(spy).toHaveBeenCalled();
    });
  });
  describe('goBack', () => {
    it('should navigate when goBack callback', () => {
      const instance = wrapper.instance();
      instance.goBack();
      expect(props.navigation.goBack).toHaveBeenCalledTimes(1);
    });
  });
  describe('change text', () => {
    it('should change state onTextChanged', () => {
      const instance = wrapper.instance();
      instance.onTextChanged('EMAIL', 'aa@aa.aa');
      expect(instance.state.email).toEqual('aa@aa.aa');
    });
  });
  afterAll(() => {
    wrapper.instance().prototype.onSendLink.mockRestore();
  });
});
