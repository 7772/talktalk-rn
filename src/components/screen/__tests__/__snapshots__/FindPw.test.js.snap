// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Shared
    isDarkContent={false}
  />
  <ScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginTop": 83.33333333333331,
          "width": "78%",
        }
      }
    >
      <Shared
        isPassword={true}
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 16.666666666666664,
          }
        }
        txt=""
        txtHint="Email"
        txtLabel="Email"
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "width": "100%",
          }
        }
      >
        <Button
          activeOpacity={0.5}
          id="send_link"
          imgLeftStyle={
            Object {
              "height": 24,
              "left": 16,
              "position": "absolute",
              "width": 24,
            }
          }
          indicatorColor="white"
          isDisabled={false}
          isLoading={false}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(58,139,255)",
              "borderColor": "rgb(58,139,255)",
              "borderRadius": 8.333333333333332,
              "borderWidth": 2.083333333333333,
              "height": 124.99999999999999,
              "justifyContent": "center",
              "marginBottom": 99.99999999999999,
              "marginLeft": 8.333333333333332,
              "marginTop": 49.99999999999999,
              "shadowColor": "rgb(58,139,255)",
              "shadowOffset": Object {
                "height": 20.83333333333333,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 8.333333333333332,
              "width": 283.3333333333333,
            }
          }
          textStyle={
            Object {
              "color": "white",
              "fontSize": 33.33333333333333,
              "fontWeight": "bold",
            }
          }
        >
          Send Link
        </Button>
      </View>
    </View>
  </ScrollView>
</View>
`;
